src/Apps/Order/Routes/Shipping2/AddressModal2.jest.tsx

```diff
@@ -0,0 +1,340 @@
+  // Cast props here because we don't actually need the query, just the relay
+  // test wrapper. -
+  // props must be passed via getWrapper's componentProps arg
+  Component: (props: unknown) => {
+    return (
+      <ShippingContext.Provider value={useComputeShippingContext({} as any)}>
+        <AddressModal {...(props as Props)} />
+      </ShippingContext.Provider>
+    )
+  },
+  query: graphql`
+    query AddressModal2TestQuery @relay_test_operation {
+      _unused: artist(id: "whocare") {
+        name
+      }
+    }
+  `,
+})
+
+describe("AddressModal", () => {
+  beforeEach(() => {
+    testAddressModalProps = {
+      onSuccess: jest.fn(),
+      modalAction: {
+        type: AddressModalActionType.EDIT_USER_ADDRESS,
+        address: savedAddress,
+      },
+
+      closeModal: jest.fn(),
+    }
+    ;(useSystemContext as jest.Mock).mockImplementation(() => {
+      return {
+        user: { lab_features: [] },
+        isLoggedIn: true,
+        relayEnvironment: mockRelayEnv,
+      }
+    })
+  })
+
+  it("renders EditModal with the title, input fields and buttons", () => {
+    const { wrapper } = getWrapper({}, testAddressModalProps, mockRelayEnv)
+
+    expect(wrapper.text()).toContain("Edit address")
+    expect(wrapper.find("input").length).toBe(7)
+    expect(wrapper.find("select").length).toBe(1)
+
+    expect(wrapper.find("Checkbox[data-test='setAsDefault']").length).toBe(1)
+    expect(wrapper.find("Clickable[data-test='deleteButton']").length).toBe(1)
+    expect(wrapper.find("Button[data-test='saveButton']").length).toBe(1)
+  })
+
+  it("renders EditModal without checkbox when address is default", () => {
+    const { wrapper } = getWrapper(
+      {},
+      {
+        ...testAddressModalProps,
+        modalAction: {
+          type: AddressModalActionType.EDIT_USER_ADDRESS,
+          address: {
+            ...savedAddress,
+            isDefault: true,
+          },
+        },
+      },
+      mockRelayEnv
+    )
+    expect(wrapper.text()).toContain("Edit address")
+    expect(wrapper.find("Checkbox[data-test='setAsDefault']").length).toBe(0)
+  })
+
+  it("renders AddModal with the title, input fields, checkbox and button", () => {
+    const { wrapper } = getWrapper(
+      {},
+      {
+        ...testAddressModalProps,
+        modalAction: {
+          type: AddressModalActionType.CREATE_USER_ADDRESS,
+        },
+      },
+      mockRelayEnv
+    )
+    expect(wrapper.text()).toContain("Add address")
+    expect(wrapper.find("input").length).toBe(7)
+    expect(wrapper.find("select").length).toBe(1)
+
+    expect(wrapper.find("Checkbox[data-test='setAsDefault']").length).toBe(1)
+    expect(wrapper.find("Clickable[data-test='deleteButton']").length).toBe(0)
+    expect(wrapper.find("Button[data-test='saveButton']").length).toBe(1)
+  })
+
+  it("clicking the delete button spawns a correct dialog", () => {
+    const { wrapper } = getWrapper({}, testAddressModalProps, mockRelayEnv)
```

Comment:
ditto here -- `testAddressModalProps` can be another default arg, so it doesn't need to be passed every time:

```tsx
const getWrapper = ({
  addressModalProps = defaultAddressModalProps,
  relayEnvironment = mockRelayEnvironment,
  ...props 
}) => {
  ...
}
```